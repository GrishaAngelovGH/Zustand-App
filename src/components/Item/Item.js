import useStore from "hooks/useStore"

const Item = ({ id, title, quantity, marked }) => {
  const incrementQty = useStore((state) => state.incrementQty)
  const decrementQty = useStore((state) => state.decrementQty)
  const markItem = useStore((state) => state.markItem)

  const handleIncrement = () => {
    incrementQty(id)
  }

  const handleDecrement = () => {
    decrementQty(id)
  }

  const handleMarkItem = () => {
    markItem(id)
  }

  return (
    <div className="col-md-2 border border-3 border-primary rounded p-2 shadow bg-light text-center">
      <p className="fw-bold">Title: {title}</p>
      <p className="fw-bold">Quantity: {quantity}</p>
      <p className="fw-bold">Marked: {`${marked}`}</p>

      <div className="row justify-content-around mt-3">
        <div className="col-md-5">
          <button
            className="btn btn-primary w-100 fs-3"
            onClick={handleIncrement}
          >
            +
          </button>
        </div>
        <div className="col-md-5">
          <button
            className="btn btn-primary w-100 fs-3"
            onClick={handleDecrement}
          >
            -
          </button>
        </div>
      </div>

      <div className="row mt-3">
        <div className="col-md-12">
          <button
            className={`btn ${marked ? "btn-info text-white" : "btn-primary"} w-100 fs-3`}
            onClick={handleMarkItem}
          >
            {marked ? "Unmark" : "Mark"}
          </button>
        </div>
      </div>
    </div>
  )
}

export default Item