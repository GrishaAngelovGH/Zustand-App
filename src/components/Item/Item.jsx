import useStore from "hooks/useStore"

const Item = ({ id, title, quantity, marked }) => {
  const incrementQty = useStore((state) => state.incrementQty)
  const decrementQty = useStore((state) => state.decrementQty)
  const markItem = useStore((state) => state.markItem)

  const handleIncrement = () => {
    incrementQty(id)
  }

  const handleDecrement = () => {
    decrementQty(id)
  }

  const handleMarkItem = () => {
    markItem(id)
  }

  return (
    <div className="border-3 border-blue-500 rounded-md p-2 shadow text-center">
      <ul className="bg-white border rounded-md text-2xl text-gray-600">
        <li className="border-b border-gray-600">Title: {title}</li>
        <li className="border-b border-gray-600">Quantity: {quantity}</li>
        <li>Marked: {`${marked}`}</li>
      </ul>

      <div className="flex gap-4 mt-3">
        <button
          className="bg-blue-500 text-white text-2xl rounded-md w-[100px] cursor-pointer"
          onClick={handleIncrement}
        >
          +
        </button>
        <button
          className="bg-blue-500 text-white text-2xl rounded-md w-[100px] cursor-pointer"
          onClick={handleDecrement}
        >
          -
        </button>
      </div>

      <div className="mt-3">
        <button
          className={`text-white rounded-md ${marked ? "bg-blue-400" : "bg-blue-500"} w-[100px] text-2xl cursor-pointer`}
          onClick={handleMarkItem}
        >
          {marked ? "Unmark" : "Mark"}
        </button>
      </div>
    </div>
  )
}

export default Item