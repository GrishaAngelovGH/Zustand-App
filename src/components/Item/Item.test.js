import { render } from '@testing-library/react'

import Item from './Item'

test('should render Item component', () => {
  const view = render(
    <Item
      title='Item 1'
      quantity={10}
      marked={false}
    />
  )

  expect(view).toMatchSnapshot()
})