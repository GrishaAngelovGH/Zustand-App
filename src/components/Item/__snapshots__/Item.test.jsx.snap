// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render Item component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="border-3 border-blue-500 rounded-md p-2 shadow text-center"
      >
        <ul
          class="bg-white border rounded-md text-2xl text-gray-600"
        >
          <li
            class="border-b border-gray-600"
          >
            Title: 
            Item 1
          </li>
          <li
            class="border-b border-gray-600"
          >
            Quantity: 
            10
          </li>
          <li>
            Marked: 
            false
          </li>
        </ul>
        <div
          class="flex gap-4 mt-3"
        >
          <button
            class="bg-blue-500 text-white text-2xl rounded-md w-[100px] cursor-pointer"
          >
            +
          </button>
          <button
            class="bg-blue-500 text-white text-2xl rounded-md w-[100px] cursor-pointer"
          >
            -
          </button>
        </div>
        <div
          class="mt-3"
        >
          <button
            class="text-white rounded-md bg-blue-500 w-[100px] text-2xl cursor-pointer"
          >
            Mark
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="border-3 border-blue-500 rounded-md p-2 shadow text-center"
    >
      <ul
        class="bg-white border rounded-md text-2xl text-gray-600"
      >
        <li
          class="border-b border-gray-600"
        >
          Title: 
          Item 1
        </li>
        <li
          class="border-b border-gray-600"
        >
          Quantity: 
          10
        </li>
        <li>
          Marked: 
          false
        </li>
      </ul>
      <div
        class="flex gap-4 mt-3"
      >
        <button
          class="bg-blue-500 text-white text-2xl rounded-md w-[100px] cursor-pointer"
        >
          +
        </button>
        <button
          class="bg-blue-500 text-white text-2xl rounded-md w-[100px] cursor-pointer"
        >
          -
        </button>
      </div>
      <div
        class="mt-3"
      >
        <button
          class="text-white rounded-md bg-blue-500 w-[100px] text-2xl cursor-pointer"
        >
          Mark
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
